src/typechecker/FStar.TypeChecker.NBE.fs

```diff
@@ -500,7 +500,7 @@ let rec translate (cfg:config) (bs:list<t>) (e:term) : t =
       debug (fun () -> BU.print2 "Application: %s @ %s\n" (P.term_to_string head) (P.args_to_string args));
       iapp cfg (translate cfg bs head) (List.map (fun x -> (translate cfg bs (fst x), snd x)) args) // Zoe : TODO avoid translation pass for args
 
-    | Tm_match(scrut, branches) ->
+    | Tm_match(scrut, _, branches) ->
```

Comment:
This would drop the returns clause as in Tm_ascribed ... maybe the resuting term would be untypable

Reply:
We should probably add a thunk to reconstruct the returns clause here (or make the thunk here return a pair)

  | Match of
       // 1. the scrutinee
       t *
       // 2. reconstructs the pattern matching
       (unit -> list<branch>)

Reply:
Fixed.